api/app/routers/morecast_v2.py

```diff
@@ -120,7 +120,7 @@ async def save_forecasts(forecasts: MoreCastForecastRequest, response: Response,
                 clear_cache_matching(station_id)
         except Exception as exc:
             logger.error("Encountered error posting forecast data to WF1 API", exc_info=exc)
-            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Error submitting forecast(s) to WF1")
+            raise WF1_HTTP_ERROR
```

Comment:
Wondering if we should raise `WF1_HTTP_ERROR` in two places or just catch it here

`except WF1_HTTP_ERROR as exc:`...

Reply:
We want to catch any exception here, not just `WF1_HTTP_ERROR`. We log the full exception for our own troubleshooting but re-raise `WF1_HTTP_ERROR` so they client knows it's an unrecoverable error on our end and receive the same instructions to attempt to retry.